### PR TITLE
i18n(zh): translate upgrade-to/v2.mdx

### PR DESCRIPTION
**Pull request by 4pmtong**
#### What kind of changes does this PR include?

- Translated content

#### Description

Translate upgrade-to/v2.mdx to chinese

### CODE REVIEW

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 1]
```diff
@@ -0,0 +1,420 @@
...
+Astro v2.0将所有构建输出资源的位置移动并统一到一个新的 `_astro/` 文件夹。
+
+<FileTree>
+- dist/
+  - _astro
+    - client.9218e799.js
+    - index.df3f880e0.css
+</FileTree>
+
+您可以通过 [新的 `build.assets` 配置项](/en/reference/configuration-reference/#buildassets) 来控制这个位置。
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:119]:
```suggestion
您可以通过 [新的 `build.assets` 配置项](/zh-cn/reference/configuration-reference/#buildassets) 来控制这个位置。
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 2]
```diff
@@ -0,0 +1,420 @@
...
+description: 如何将你的项目升级到 Astro 的最新版本.
+i18nReady: true
+---
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+import FileTree from '~/components/FileTree.astro'
+
+
+本指南将帮助您从 Astro v1 迁移到 Astro v2。
+
+需要将旧项目升级到 v1 吗？请参阅我们的 [旧版本迁移指南](/en/guides/upgrade-to/v1/).
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:12]:
```suggestion
需要将旧项目升级到 v1 吗？请参阅我们的 [旧版本迁移指南](/zh-cn/guides/upgrade-to/v1/).
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+1. 使用以下命令检查本地版本的 Node：
+
+    ```sh
+    node -v
+ 
+    ```
+    如果你的本地开发环境需要升级，[安装 Node](https://nodejs.org/en/download/).
+
+
+2. 查看您的 [部署环境](/en/guides/deploy/) 的文档，以确认它们是否支持 Node 16。
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:73]:
```suggestion
2. 查看您的 [部署环境](/zh-cn/guides/deploy/) 的文档，以确认它们是否支持 Node 16。
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 4]
```diff
@@ -0,0 +1,420 @@
...
+
+    您可以在仪表板配置设置或 `.nvmrc` 文件中为您的Astro项目指定 `Node 16.12.0`。
+
+
+### 保留：`src/content/`
+
+Astro v2.0 现在包含了用于将Markdown和MDX文件整理到 [内容集合](/en/guides/content-collections/) 中的Collections API。此API将`src/content/`设定为一个特殊的文件夹。
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:80]:
```suggestion
Astro v2.0 现在包含了用于将Markdown和MDX文件整理到 [内容集合](/zh-cn/guides/content-collections/) 中的Collections API。此API将`src/content/`设定为一个特殊的文件夹。
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 5]
```diff
@@ -0,0 +1,420 @@
...
+
+#### 我应该怎么做？
+
+重命名现有的`src/content/`文件夹以避免冲突。 如果该文件夹存在，现在只能用于 [内容集合](/en/guides/content-collections/).
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:84]:
```suggestion
重命名现有的`src/content/`文件夹以避免冲突。 如果该文件夹存在，现在只能用于 [内容集合](/zh-cn/guides/content-collections/).
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 6]
```diff
@@ -0,0 +1,420 @@
...
+
+### 变更：MDX 插件配置
+
+#### 替换：`extendPlugins` 更改为 `extendMarkdownConfig`
+
+在 v1.x 中，MDX 集成的 `extendPlugins` 选项管理了你的 MDX 文件应如何继承你的 Markdown 配置：全部继承你的 Markdown 配置(`markdown`)，或者只继承 Astro 的默认插件(`default`)。
+
+Astro v2.0 用三个新的、独立可配置的选项替换了 `mdx.extendPlugins` 控制的行为，这些选项默认都为 `true`：
+
+- **[`mdx.extendMarkdownConfig`](/en/guides/integrations-guide/mdx/#extendmarkdownconfig)** 继承全部或不继承你的 Markdown 配置
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:185]:
```suggestion
- **[`mdx.extendMarkdownConfig`](/zh-cn/guides/integrations-guide/mdx/#extendmarkdownconfig)** 继承全部或不继承你的 Markdown 配置
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 7]
```diff
@@ -0,0 +1,420 @@
...
+      smartypants: true,
+      gfm: true,
+    }),
+  ],
+});
+```
+
+#### 新增：更多的 MDX 配置选项以匹配 Markdown
+
+Astro v2.0 现在允许您在 MDX 集成配置中分别设置 [每一个可用的 Markdown 配置选项](/en/reference/configuration-reference/#markdown-options)（除了 `drafts`）。
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:228]:
```suggestion
Astro v2.0 现在允许您在 MDX 集成配置中分别设置 [每一个可用的 Markdown 配置选项](/zh-cn/reference/configuration-reference/#markdown-options)（除了 `drafts`）。
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 8]
```diff
@@ -0,0 +1,420 @@
...
+    items: await pagesGlobToRssItems(
+      import.meta.glob('./blog/*.{md,mdx}'),
+    ),
+  });
+}
+```
+
+### 变更：Svelte IDE 支持
+
+如果你正在使用 [`@astrojs/svelte` 集成](/en/guides/integrations-guide/svelte/)，Astro v2.0 需要在你的项目中有一个 `svelte.config.js` 文件。这是为了提供 IDE 的自动补全功能。
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:291]:
```suggestion
如果你正在使用 [`@astrojs/svelte` 集成](/zh-cn/guides/integrations-guide/svelte/)，Astro v2.0 需要在你的项目中有一个 `svelte.config.js` 文件。这是为了提供 IDE 的自动补全功能。
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 9]
```diff
@@ -0,0 +1,420 @@
...
+// astro.config.mjs
+export default defineConfig({
+	legacy: {
+		astroFlavoredMarkdown: true,
+	},
+})
+
+```
+
+如果你在 v1.x 版本中使用了这个特性，我们推荐 [使用 MDX 集成](/en/guides/integrations-guide/mdx/)，它允许你将组件和 JSX 表达式与 Markdown 语法结合起来。
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:328]:
```suggestion
如果你在 v1.x 版本中使用了这个特性，我们推荐 [使用 MDX 集成](/zh-cn/guides/integrations-guide/mdx/)，它允许你将组件和 JSX 表达式与 Markdown 语法结合起来。
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 10]
```diff
@@ -0,0 +1,420 @@
...
+
+### 移除: `Astro.fetchContent()`
+
+在 v0.26 版本中，Astro 废弃了用于从本地 Markdown 文件中获取数据的 `Astro.fetchContent()` 方法。
+
+Astro v2.0 完全移除了这个选项。在你的代码中使用 `Astro.fetchContent()` 会导致错误。
+
+#### 我应该怎么做？
+
+使用 [`Astro.glob()`](/en/guides/imports/#astroglob) 方法获取 Markdown 文件，或者转换为 [内容集合](/en/guides/content-collections/) 功能。
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:358]:
```suggestion
使用 [`Astro.glob()`](/zh-cn/guides/imports/#astroglob) 方法获取 Markdown 文件，或者转换为 [内容集合](/zh-cn/guides/content-collections/) 功能。
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 11]
```diff
@@ -0,0 +1,420 @@
...
+    contentCollections: true,
+    prerender: true,
+    errorOverlay: true,
+  },
+})
+```
+
+这些功能现在已默认可用：
+
+- [内容集合](/en/guides/content-collections/)，用于以类型安全的方式管理你的 Markdown 和 MDX 文件。
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:414]:
```suggestion
- [内容集合](/zh-cn/guides/content-collections/)，用于以类型安全的方式管理你的 Markdown 和 MDX 文件。
```

File: src/content/docs/zh-cn/guides/upgrade-to/v2.mdx [hunk 12]
```diff
@@ -0,0 +1,420 @@
...
+- 在使用 SSR 时，将个别页面[预渲染为静态 HTML](/en/guides/server-side-rendering/#configuring-individual-routes)，以提高速度和缓存能力。详见 配置个别路由。
```
Comment (luoingly) [inline src/content/docs/zh-cn/guides/upgrade-to/v2.mdx:415]:
```suggestion
- 在使用 SSR 时，将个别页面[预渲染为静态 HTML](/zh-cn/guides/server-side-rendering/#configuring-individual-routes)，以提高速度和缓存能力。详见 配置个别路由。
```

**Review by luoingly — CHANGES_REQUESTED** [review]
Thank you for your contributions to this project! 💜

Although I think there are several details that need to be modified in your translation, overall I think it is fine, except for the issues I have pointed out (suggested changes):

You should change all the `/en/` in the internal hyperlinks of the document to `/zh-cn/`, and you don't need to worry about whether the target page has been translated.

Additionally, if you could add spaces between Chinese and English, while removing unnecessary spaces between Chinese text (without affecting Markdown rendering), I think it would be better! 🚀

**Comment by Yan-Thomas** [comment]
Thanks for this @4pmtong! Could you take a look at luoingly's suggestions and apply them if they make sense? I'd love to get this merged 🙌

**Comment by Yan-Thomas** [comment]
I see this PR got stale, no worries! I'll be merging it in its current state with the suggestions applied. If you have any more suggestions for improvements, feel free to make another PR and I'll take a look into it! 🙌

**Review by Yan-Thomas — APPROVED** [review]
LGTM! Thanks @4pmtong and @luoingly for the careful review! 🙌

**Comment by 4pmtong** [comment]
My apologies for the delay in responding as I was without internet access these past few days. I've just seen your message. Thanks @Yan-Thomas and @luoingly  for your review and suggestions!

**Pull request merged**